### PR TITLE
fetch game data and shove into canvas

### PR DESCRIPTION
**Pull request by josephcwalker**

### CODE REVIEW

File: src/state/state_manager.py [hunk 1]
```diff
@@ -1,12 +1,7 @@
 class StateManager:
     def __init__(self):
         self.last_message = None
-
-    def set_last_message(self, message):
-        self.last_message = message
-
-    def get_last_message(self):
-        return self.last_message
+        self.canvas = None
```
Comment (AyaPK) [inline src/state/state_manager.py:8]: I like the refactor here, just make sure to update the places where `set_last_message()` and `get_last_message()` are used

Comment (josephcwalker) [reply src/state/state_manager.py:8]: truuu
done

File: src/integration/pyboy_integration.py [hunk 2]
```diff
@@ -28,4 +30,41 @@ def pyboy_tick(button=""):
...
+
+Name: {self.player_name}
+"""
+
+
+STRING_TERMINATOR = 80
+ASCII_DELTA = 63
+
+
+def _bytes_as_gen1_string(data) -> str:
```
Comment (AyaPK) [inline src/integration/pyboy_integration.py]: Reckon we could do some sick list comprehension here

```
def _bytes_as_gen1_string(data) -> str:
    return ''.join(chr(byte - ASCII_DELTA) for byte in data if byte != STRING_TERMINATOR)
```

Comment (josephcwalker) [reply src/integration/pyboy_integration.py]: yeah this was ripped directly from the PR you found

File: src/integration/pyboy_integration.py [hunk 3]
```diff
@@ -28,4 +32,128 @@ def pyboy_tick(button=""):
...
+
+    game_info = GameInformation.from_pyboy(pyboy)
+
     pyboy.stop()
+
+    return game_info
+
+
+@dataclass
+class Pokemon:
```
Comment (AyaPK) [inline src/integration/pyboy_integration.py]: I wonder if we should be moving this in to a new `gen_1_integration.py` file, to keep it separate from generic pyboy stuff

Comment (josephcwalker) [reply src/integration/pyboy_integration.py]: done

File: src/bot/slack_bot.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def start_timer(client, say, event):
     time.sleep(TIMER_DURATION)
     print(f"{TIMER_DURATION} second timer complete!")
 
+    timer_active = False
```
Comment (josephcwalker) [inline src/bot/slack_bot.py:54]: i've moved this here because currently if you react while the reacts are being populated, it doesn't count

**Pull request merged**